File: backend/schemas/notifications.py
import uuid
from datetime import datetime
from pydantic import BaseModel, ConfigDict


class NotificationOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    user_id: uuid.UUID
    type: str
    title: str
    body: str
    read: bool
    sent_email: bool
    created_at: datetime

File: frontend/src/schemas/notifications.ts
import { z } from 'zod'

export const NotificationSchema = z.object({
  id: z.string().uuid(),
  user_id: z.string().uuid(),
  type: z.string(),
  title: z.string(),
  body: z.string(),
  read: z.boolean(),
  sent_email: z.boolean(),
  created_at: z.string(),
})

export type Notification = z.infer<typeof NotificationSchema>

export const NotificationListSchema = z.array(NotificationSchema)
